### PR TITLE
Multi factorial

### PR DESCRIPTION
**Pull request by irevoire**
Hey, currently numbat doesn't handle multifactorial correctly, see the following screenshot:
image

This should in fact returns (5 * 3 * 1): https://en.wikipedia.org/wiki/Double_factorial#Generalizations

This PR adds support for it, it's a breaking change though since numbat was already accepting the expression.

### CODE REVIEW

File: numbat/tests/interpreter.rs [hunk 1]
```diff
@@ -169,6 +169,12 @@ fn test_factorial() {
     expect_output("-5!", "-120");
     expect_output("-(5!)", "-120");
     expect_output("-(5)!", "-120");
+    expect_output("5!!", "15");
```
Comment (sharkdp) [inline numbat/tests/interpreter.rs:172]: Can we expand at least the double-factorial tests a bit more and include some edge cases, as well as an example for even numbers. And maybe we could write the tests in a way that it's easy to see that we get the desired result. By moving them to a Numbat-source test suite? Something like
```
assert_eq(7!!, 7 * 5 * 3 * 1)
assert_eq(8!!, 8 * 4 * 2)
```

**Review by sharkdp — APPROVED** [review]
Thank you very much, this looks great. I'm not concerned about the breaking change.

File: numbat/tests/interpreter.rs (hunk 1 above)
Comment (irevoire) [reply numbat/tests/interpreter.rs:177]: Hey, that should be done in https://github.com/sharkdp/numbat/pull/695/commits/fb066f0e9c82c42c4b6be3d2db1a75c810157fb4

Could you confirm me that we cannot write an assert in a function in numbat currently?
I wanted to write a more « procedural » test with a recursive function that assert at each step that the function works on all numbers from 0 to 10 with multiple orders but then I didn’t see how to do it 🤔

Comment (Goju-Ryu) [reply numbat/tests/interpreter.rs:177]: I can confirm that it isn't currently possible. I wanted to make a function to encapsulate a repeating pattern in the tests I wrote and learned it wasn't possible yet. 
There was some talk about the possibility of numbat having macros for such cases, but that's still just an idea.

**Review by sharkdp — APPROVED** [review]
Thank you
